solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -375,10 +374,152 @@ private StatefulCollectionWatch compute(
         BiFunction<String, StatefulCollectionWatch, StatefulCollectionWatch> remappingFunction) {
       return statefulWatchesByCollectionName.compute(collectionName, remappingFunction);
     }
+
+    public boolean isWatched(String coll) {
+      return statefulWatchesByCollectionName.containsKey(coll);
+    }
   }
 
-  private static class StatefulCollectionWatch extends CollectionWatch<DocCollectionWatcher> {
+  private class StatefulCollectionWatch extends CollectionWatch<DocCollectionWatcher> {
     private DocCollection currentState;
+
+    Watcher persistentWatcher;
```

Comment:
init of this uses double-checked locking later on https://github.com/cowpaths/fullstory-solr/pull/105/files#diff-9b1918d641f78a8c245306b47ffd7a4554a12771ed4cb42293a411537fe7b54dR1563-R1568. Perhaps add `volatile`? :)